src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt

```diff
@@ -78,27 +87,43 @@ fun zipDirectory(directory: File): ByteArray {
     return byteOutput.toByteArray()
 }
 
-class ResultReceiver() : CompilerFleetMessageReceiver() {
+fun getTaskId(data: ByteArray): String {
+    val messageDigest = MessageDigest.getInstance("SHA-1")
+
+    return String(messageDigest.digest(data))
+}
+
+private val LOGGER = LoggerFactory.getLogger("ResultReceiver")
+
+class ResultReceiver(private val rootFile: File) : CompilerFleetMessageReceiver() {
     override fun processMessage(message: PubsubMessage) {
         val result = CompilerFleet.CompilerFleetResult.parseFrom(message.data)
 
-        println(result.taskId)
-        println(String(result.resultBytes.toByteArray()))
+        val expectedTaskId = getTaskId(rootFile.name.toByteArray())
+        if (result.taskId != expectedTaskId) {
+            LOGGER.error("Task ids don't match: \nexpected:{}, \nactual:{}", expectedTaskId, result.taskId)
+            return
+        }
+
+        val file = File(FilenameUtils.removeExtension(rootFile.absolutePath) + PDF_EXTENSION)
```

Comment:
Did you try testing when project directory is not in the current working directory? E.g. when you run cli with `--dir ../../project` ? This code should fail because of no such directory ./foo/bar/dir1

This CLI is just for demo purposes, so we basically do not care about it too much, but even for demo purposes storing the result next to input will make you sending more data the next time you run your CLI app (it will add PDF to ZIP).

Reply:
Okay, I've got the point. 
At the moment, I'm just creating pdf file in project root.